src/shared/components/data-table/DataTable.tsx

```diff
@@ -63,13 +63,17 @@ const DataTable = (props: TableProps) => {
     ...(props.state || {})
   };
 
+  const firstRenderRef = useRef(true);
   const [tableState, dispatch] = useReducer(
     tableReducer,
     initialDataTableState
   );
 
   useEffect(() => {
-    props.onStateChange?.(tableState);
+    if (!firstRenderRef.current) {
+      props.onStateChange?.(tableState);
+    }
+    firstRenderRef.current = false;
   }, [tableState]);
```

Comment:
Not liking this idea, with useEffect and the firstRenderRef.

Could we pull off something like this: create a wrapper around the reducer function, which behaves like the reducer, but also notifies the parent of the updated state:

```ts
// use this in the useReducer instead of the tableReducer
const reducerWithCallback: typeof tableReducer = (state, action) => {
  const newState = tableReducer(state, action);

  // if there are some actions when we want not to notify the parent
  // (e.g. if the state has updated because the parent said so),
  // then skip the callback. Plus, don't notify parent if the state hasn't changed.
  if (newState !== state) {
    props?.onStateChange(newState);  
  }

  return newState;
}
```